fixtures/dom/src/components/fixtures/index.js

```diff
@@ -1,62 +1,53 @@
-import SelectFixtures from './selects';
-import TextAreaFixtures from './textareas';
-import InputChangeEvents from './input-change-events';
-import NumberInputFixtures from './number-inputs';
-import PasswordInputFixtures from './password-inputs';
-import ButtonFixtures from './buttons';
-import DateInputFixtures from './date-inputs';
-import ErrorHandling from './error-handling';
-import EventPooling from './event-pooling';
-import CustomElementFixtures from './custom-elements';
-import MediaEventsFixtures from './media-events';
-import PointerEventsFixtures from './pointer-events';
-import MouseEventsFixtures from './mouse-events';
-import SelectionEventsFixtures from './selection-events';
-
 const React = window.React;
+const fixturePath = window.location.pathname;
 
 /**
  * A simple routing component that renders the appropriate
  * fixture based on the location pathname.
  */
-function FixturesPage() {
-  switch (window.location.pathname) {
-    case '/text-inputs':
-      return <TextInputFixtures />;
-    case '/range-inputs':
-      return <RangeInputFixtures />;
-    case '/selects':
-      return <SelectFixtures />;
-    case '/textareas':
-      return <TextAreaFixtures />;
-    case '/input-change-events':
-      return <InputChangeEvents />;
-    case '/number-inputs':
-      return <NumberInputFixtures />;
-    case '/password-inputs':
-      return <PasswordInputFixtures />;
-    case '/buttons':
-      return <ButtonFixtures />;
-    case '/date-inputs':
-      return <DateInputFixtures />;
-    case '/error-handling':
-      return <ErrorHandling />;
-    case '/event-pooling':
-      return <EventPooling />;
-    case '/custom-elements':
-      return <CustomElementFixtures />;
-    case '/media-events':
-      return <MediaEventsFixtures />;
-    case '/pointer-events':
-      return <PointerEventsFixtures />;
-    case '/mouse-events':
-      return <MouseEventsFixtures />;
-    case '/selection-events':
-      return <SelectionEventsFixtures />;
-    default:
-      return <p>Please select a test fixture.</p>;
+class FixturesPage extends React.Component {
+  static defaultProps = {
+    fixturePath: fixturePath === '/' ? '/home' : fixturePath,
+  };
+
+  state = {
+    isLoading: true,
+    error: null,
+    Fixture: null,
+  };
+
+  componentDidMount() {
+    this.loadFixture();
+  }
+
+  async loadFixture() {
+    const {fixturePath} = this.props;
+
+    try {
+      let module = await import(`.${fixturePath}`);
+
+      this.setState({Fixture: module.default});
+    } catch (error) {
+      console.error(error);
+      this.setState({error});
+    } finally {
+      this.setState({isLoading: false});
+    }
+  }
+
+  render() {
+    const {fixturePath} = this.props;
+    const {Fixture, error, isLoading} = this.state;
+
+    if (isLoading) {
+      return <p>Awaiting fixture...</p>;
+    }
+
+    if (error) {
+      return <p>Fixture at path {fixturePath} could not be loaded.</p>;
```

Comment:
Include `error.message`?

Reply:
Got it, 69b89ca8d

![screenshot 2018-08-11 at 2 43 37 pm](https://user-images.githubusercontent.com/590904/43996318-2815341e-9d75-11e8-89b9-f1d36b47b354.png)

Reply:
I debated back and forth with including the stack trace too, but it felt noisy when this is more or less a 404 every time